project/SparkBuild.scala

```diff
@@ -167,8 +167,8 @@ object SparkBuild extends PomBuild {
     publishLocalBoth <<= Seq(publishLocal in MavenCompile, publishLocal).dependOn,
 
     javacOptions in (Compile, doc) ++= {
-      val Array(major, minor, _) = System.getProperty("java.version").split("\\.", 3)
-      if (major.toInt >= 1 && minor.toInt >= 8) Seq("-Xdoclint:all", "-Xdoclint:-missing") else Seq.empty
+      val version = System.getProperty("java.version")
+      if (version >= "1.8.0") Seq("-Xdoclint:all", "-Xdoclint:-missing") else Seq.empty
```

Comment:
Hm, this only happens to work for 1.8 vs 9. This seems hacky compared to just emulating the simple logic you wrote above -- why not that?